src/queens/parameters/random_fields/karhunen_loeve.py

```diff
@@ -86,71 +89,84 @@ def __init__(
         if latent_dimension is not None:
             self.dimension = latent_dimension
         else:
-            self.dimension = None
+            self.dimension = 0
 
         self.calculate_covariance_matrix()
         self.eigendecomp_cov_matrix()
```

Comment:
Maybe this approach would be simpler:
```suggestion
        self.calculate_covariance_matrix()
        self.eigendecomp_cov_matrix(latent_dimension)
```